### PR TITLE
Sanlouise 10397 Add cancellation modal to Personal Information

### PR DESCRIPTION
**Pull request by sanlouise**
## Description
PROFILE 2.0 PERSONAL INFORMATION -- When a user has edited info in a section and then clicks 'Cancel', we want to show a modal to make sure the user knows that any unsaved info will be lost.

## Testing done
Works locally. We should add tests for the functionality of this modal when we write more comprehensive integration tests.

## Screenshots
image

## Acceptance criteria
- [x] PROFILE 2.0 PERSONAL INFORMATION Add a modal when the user cancels out of an unsaved section

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/user/profile/vet360/containers/VAPProfileField.jsx [hunk 1]
```diff
@@ -288,9 +298,37 @@ class VAPProfileField extends React.Component {
...
-          ]?.toLowerCase()}`}
+          title={'Are you sure?'}
+          status="warning"
+          visible={this.state.showConfirmCancelModal}
+          onClose={() => {
+            this.setState({ showConfirmCancelModal: false });
+          }}
+        >
+          <p>
+            {' '}
```
Comment (erikphansen) [inline src/platform/user/profile/vet360/containers/VAPProfileField.jsx:309]: is this needed?

File: src/platform/user/profile/vet360/containers/VAPProfileField.jsx [hunk 2]
```diff
@@ -74,10 +74,12 @@ class VAPProfileField extends React.Component {
 
   static defaultProps = {
     fieldName: '',
+    showConfirmCancelModal: false,
```
Comment (erikphansen) [inline src/platform/user/profile/vet360/containers/VAPProfileField.jsx:77]: `hasUnsavedEdits`?

**Review by erikphansen — APPROVED** [review]
A couple nits. My instinct is to want to only have one `Modal` but that seems entirely unnecessary!

File: src/platform/user/profile/vet360/containers/VAPProfileField.jsx (hunk 1 above)
Comment (sanlouise) [reply src/platform/user/profile/vet360/containers/VAPProfileField.jsx:309]: ESLint added in that awkward white space, I cannot save the file when I take it away :(

File: src/platform/user/profile/vet360/containers/VAPProfileField.jsx (hunk 2 above)
Comment (erikphansen) [reply src/platform/user/profile/vet360/containers/VAPProfileField.jsx:77]: sorry, I mean should the entry in `defaultProps` be `hasUnsavedEdits`?

Comment (sanlouise) [reply src/platform/user/profile/vet360/containers/VAPProfileField.jsx:77]: Haha, yes. I need to make myself some coffee ☕

File: src/platform/user/profile/vet360/containers/VAPProfileField.jsx (hunk 1 above)
Comment (bkjohnson) [reply src/platform/user/profile/vet360/containers/VAPProfileField.jsx:309]: It's a `prettier` thing :shrug:

**Pull request merged**